actionview/lib/action_view/template/types.rb

```diff
@@ -28,30 +28,30 @@ def initialize(symbol)
           @symbol = symbol.to_sym
         end
 
-        delegate :to_s, :to_sym, :to => :symbol
+        def to_s
+          @symbol.to_s
+        end
         alias to_str to_s
 
         def ref
-          to_sym || to_s
+          @symbol
         end
+        alias to_sym ref
 
         def ==(type)
-          return false if type.blank?
-          symbol.to_sym == type.to_sym
+          @symbol == type.to_sym unless type.blank?
         end
       end
 
-      cattr_accessor :type_klass
-
-      def self.delegate_to(klass)
-        self.type_klass = klass
-      end
-
-      delegate_to Type
+      cattr_accessor(:type_klass) { Type }
 
       def self.[](type)
         type_klass[type]
       end
+
+      def symbols
+        type_klass::SET.symbols
```

Comment:
Just to clear something up here. Generally duck typing is preferred, so we don't have to maintain type consistency with `Mime::SET.symbols`. But I'm changing to an Array because of https://github.com/rails/rails/pull/23085/files#diff-d5d92e14e3693ca777dad370615cdaa9R70, which I've just demonstrated above will fail.